R/checks.R

```diff
@@ -124,6 +123,48 @@ check_training_frac <- function(frac) {
   }
 }
 
+#' Check the validity of the training indices
+#'
+#' @param training_inds vector of integers corresponding to samples for the training set
+#' @param dataset data frame containing the entire dataset
+#'
+#' @export
+#'
+#' @examples
+#' training_indices <- otu_small %>%
+#'   nrow() %>%
+#'   sample(., size = 160)
+#' check_training_indices(training_indices, otu_small)
+check_training_indices <- function(training_inds, dataset) {
+  if (!all(is_whole_number(training_inds))) {
+    warning(
+      "The training indices vector contains non-integer numbers."
+    )
+  }
+  stop_msg <- character(0)
+  if (min(training_inds) < 1) {
+    stop_msg <- paste0(stop_msg,
+      "The training indices vector contains a value less than 1.",
+      sep = "\n"
+    )
+  }
+  if (max(training_inds) > nrow(dataset)) {
+    stop_msg <- paste0(stop_msg,
+      "The training indices vector contains a value that is too large for the number of rows in the dataset.",
+      sep = "\n"
+    )
+  }
+  if (length(training_inds) > nrow(dataset)) {
```

Comment:
should this be >= because the whole dataset cant be in the training set right? or maybe an addtional error for length(training_inds) == nrow(dataset)

Reply:
Fixed here: 0536eb6